onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -393,7 +393,9 @@ class PlannerImpl {
       // Identify where each output of this node should be allocated.
       // This is determined by the opkernel bound to the node.
       const KernelCreateInfo* kernel_create_info = nullptr;
-      ORT_RETURN_IF_ERROR(kernel_registry_.SearchKernelRegistry(*pnode, &kernel_create_info));
+      auto st = kernel_registry_.SearchKernelRegistry(*pnode, &kernel_create_info);
```

Comment:
just curious. how do we hit this? shouldn't missing implementations be caught earlier than in planner?

Reply:
By testing ort with the latest onnx 1.7.

Reply:
SearchKernelRegistry returns FAIL if the node was not assigned to any EP. This line would report this event incorrectly as NOT_IMPLEMENTED. Should we instead modify SearchKernelRegistry to return NOT_IMPLEMENTED for this scenario and keep ORT_RETURN_IF_ERROR here?
```
  std::ostringstream errormsg;
  errormsg << "Failed to find kernel for " << node.OpType();
  if (node.Op() != nullptr) errormsg << "(" << node.Op()->since_version() << ")";
  if (!node.Name().empty()) errormsg << " (node " << node.Name() << ")";
  return Status(ONNXRUNTIME, FAIL, errormsg.str());
```